flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/AllToAllBlockingResultInfo.java

```diff
@@ -139,7 +153,14 @@ public void resetPartitionInfo(int partitionIndex) {
     }
 
     public List<Long> getAggregatedSubpartitionBytes() {
-        checkState(aggregatedSubpartitionBytes != null, "Not all partition infos are ready");
-        return Collections.unmodifiableList(aggregatedSubpartitionBytes);
+        checkState(
+                aggregatedSubpartitionBytes != null
+                        || subpartitionBytesByPartitionIndex.size() == numOfPartitions,
+                "Not all partition infos are ready");
+        if (aggregatedSubpartitionBytes == null) {
+            return getAggregatedSubpartitionBytesInternal();
```

Comment:
Can it just invoke `aggregateSubpartitionBytes()`? Invoking `getAggregatedSubpartitionBytesInternal()` will not update the `aggregatedSubpartitionBytes` nor clear the `subpartitionBytesByPartitionIndex`. 
This comment also applies to other places where `getAggregatedSubpartitionBytesInternal()` is invoked.

Reply:
Here, I use `getAggregatedSubpartitionBytesInternal` because I do not want to clear `subpartitionBytesByPartitionIndex` and set `aggregatedSubpartitionBytes` since it does not meet the condition for aggregation.

So, I introduced the method `getAggregatedSubpartitionBytesInternal`, which is only for querying and not for updating any state.

And I will add some comments for explaining.

Reply:
Maybe explain in what the case the aggregated result will be needed while the fined grained stats should not be aggregated.